src/Adapter/MSTest.TestAdapter/ObjectModel/UnitTestElement.cs

```diff
@@ -195,7 +195,10 @@ internal TestCase ToTestCase()
 
             testCase.SetPropertyValue(Constants.TestDynamicDataTypeProperty, (int)TestMethod.DataType);
             testCase.SetPropertyValue(Constants.TestDynamicDataProperty, data);
-            testCase.SetPropertyValue(Constants.TestDataSourceIgnoreMessageProperty, TestMethod.TestDataSourceIgnoreMessage);
+            if (TestMethod.TestDataSourceIgnoreMessage is not null)
```

Comment:
```suggestion
            // VSTest serialization doesn't handle null so instead don't set the property so that it's deserialized as null
            if (TestMethod.TestDataSourceIgnoreMessage is not null)
```